DataConnectors/GoogleWorkspaceReports/GWorkspaceReportsAPISentinelConnector/__init__.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import print_function
+import pickle
+from googleapiclient.discovery import build
+import datetime
+import json
+import base64
+import hashlib
+import hmac
+import requests
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+import azure.functions as func
+import logging
+import os
+import time
+
+customer_id = os.environ['WorkspaceID'] 
+shared_key = os.environ['WorkspaceKey']
+pickle_str = os.environ['GooglePickleString']
+pickle_string = base64.b64decode(pickle_str)
+SCOPES = ['https://www.googleapis.com/auth/admin.reports.audit.readonly']
+activities = ["login", "calendar", "drive", "admin", "mobile", "token", "user_accounts"]
+
+def get_credentials():
+    creds = None
+    if pickle_string:
+        try:
+            creds = pickle.loads(pickle_string)
+        except Exception as pickle_read_exception:
+            print("ERROR " + str(pickle_read_exception))
+            logging.error('Error while loading pickle string: {}'.format(pickle_read_exception))
+    else:
+        print("ERROR - pickle_string is empty. Exit")
+        logging.error('Error - pickle_string is empty. Exit')
+        exit(1)
+    return creds
+
+def generate_date():
+    current_time = datetime.datetime.utcnow().replace(second=0, microsecond=0) - datetime.timedelta(minutes=10)
+    past_time = current_time - datetime.timedelta(minutes=10)
+    return (past_time.strftime("%Y-%m-%dT%H:%M:%SZ"), current_time.strftime("%Y-%m-%dT%H:%M:%SZ"))
+
+def get_result(activity,start_time, end_time):
+    result_activities = []
+    service = build('admin', 'reports_v1', credentials=creds, cache_discovery=False)
+    results = service.activities().list(userKey='all', applicationName=activity,
+                                              maxResults=1000, startTime=start_time, endTime=end_time).execute()
+    next_page_token = results.get('nextPageToken', None)
+    result = results.get('items', [])
+    result_activities.extend(result)
+    while next_page_token is not None:
+        results = service.activities().list(userKey='all', applicationName=activity,
+                                            maxResults=1000, startTime=start_time, endTime=end_time, pageToken=next_page_token).execute()
+        next_page_token = results.get('nextPageToken', None)
+        result = results.get('items', [])
+        result_activities.extend(result)
+    if result_activities == None or len(result_activities) == 0:
+        print("Logs not founded for {} activity".format(activity))
+        logging.info("Logs not founded for {} activity".format(activity))
+        logging.info("Activity - {}, {} events was processed)".format(activity, len(result_activities)))

```

Comment:
this message is a bit misleading because the events were only fetched but not yet posted to LA.
Maybe change to `"Activity - {}, processing {} events"`

Reply:
Ok, I`ll change it